backend/src/main/java/com/htecgroup/skynest/annotation/validator/EditValidator.java

```diff
@@ -0,0 +1,40 @@
+package com.htecgroup.skynest.annotation.validator;
+
+import com.htecgroup.skynest.annotation.CurrentUserCanEdit;
+import com.htecgroup.skynest.model.dto.LoggedUserDto;
+import com.htecgroup.skynest.service.CurrentUserService;
+import com.htecgroup.skynest.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import java.util.UUID;
+
+@Component
+@RequiredArgsConstructor
+public class EditValidator implements ConstraintValidator<CurrentUserCanEdit, UUID> {
```

Comment:
what is the difference between this and ViewValidator? I believe you can use the same validator, and the same annotation, just override with Worker/Admin role in its usage definition

Reply:
In ViewValidator validation is that worker cannot see any other user details than his, in EditValidator there need to be checked that if admin is trying to edit or delete another admin it cant be done. So it needs to check if the accessed uuid from param(User with that Id) is admin than his info cannot be edited. But admin can view another admin info, just cant delete or edit other than his. So thats why there are two validator classes

Reply:
I forgot to mention that he can edit or delete another worker, but not another admin. I don't see the solution to use one validator class to include this separate checks. Because for different roles view can be accessed but not editing part. Admin can see every other user but can not edit another admin.

Reply:
Maybe to add access field into annotation class and then to pass Edit or View to it so it knows witch call is made and than in validator class to have switch or something like that to do different checks based on access field value.

Reply:
I would keep everything separate.. Meaning separate validator for edit , for view and for delete..
For edit and delete shouldn't be dependent of user role, just the uuid, if I understood correctly.
For view is dependent of user role